### PR TITLE
Add Github Action for Dockerfiles to IDA cookiecutter.

### PR DESCRIPTION
**Pull request by adzuci**
**Description:**

Describe in a couple of sentences what this PR adds

**JIRA:**

[XXX-XXXX](https://openedx.atlassian.net/browse/XXX-XXXX)

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: cookiecutter-django-ida/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+COPY . /edx/app/repo_name
+WORKDIR /edx/app/repo_name
+
+ENV repo_name_CFG /edx/etc/repo_name.yml
```
Comment (adzuci) [inline cookiecutter-django-ida/Dockerfile:21]: @syed-awais-ali or @jdmulloy I put this WIP PR up, but I haven't figured out how to template out a variable in all caps and have a suspicion that I'm missing some pieces (like the `gunicorn_configuration.py` file)

**Comment by jinder1s** [comment]
@adzuci hi, warning, you are putting files in the wrong place. They should go in this folder: [/cookiecutter-django-ida/{{cookiecutter.repo_name}}/](https://github.com/edx/edx-cookiecutters/tree/master/cookiecutter-django-ida/%7B%7Bcookiecutter.repo_name%7D%7D)

File: cookiecutter-django-ida/.github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build and Push docker image
+        env:
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
```
Comment (adzuci) [inline cookiecutter-django-ida/.github/workflows/docker-publish.yml]:
```
Unable to create file '.github/workflows/docker-publish.yml'
Error message: 'secrets' is undefined
```

**Comment by adzuci** [comment]
I was able to figure out the issues I had and move the files into the right place.  `make cookiecutter-django-ida` now seems to run cleanly as well, but I'd like another set of eyes to let me know if the `.github/workflow/docker-publish.yml` file looks correct.  It's based on https://github.com/edx/course-discovery/blob/master/.github/workflows/docker-publish.yml

File: cookiecutter-django-ida/Dockerfile (hunk 1 above)
Comment (adzuci) [reply cookiecutter-django-ida/Dockerfile]: I realized that I didn't need to update the Dockerfile, so this comment no longer applies.

**Review by syed-awais-ali — APPROVED** [review]

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile [hunk 3]
```diff
@@ -184,3 +184,20 @@ db-shell: # Run the app shell as root, enter the app's database
...
+	docker build . -f Dockerfile --target newrelic -t openedx/repo_name:latest-newrelic
+
+github_docker_tag: docker_build
+	docker tag openedx/repo_name openedx/repo_name:${GITHUB_SHA}
+	docker tag openedx/repo_name:latest-newrelic openedx/repo_name:${GITHUB_SHA}-newrelic
+
+github_docker_auth:
+	echo "$$DOCKERHUB_PASSWORD" | docker login -u "$$DOCKERHUB_USERNAME" --password-stdin
+
+github_docker_push: docker_tag docker_auth ## push to docker hub
```
Comment (syed-awais-ali) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile:199]: docker tag should be replaced by github_docker_tag and docker auth with github_docker_auth

**Review by syed-awais-ali — CHANGES_REQUESTED** [review]

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile (hunk 3 above)
Comment (adzuci) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile]: Good catch, fixed.

**Pull request merged**